packages/flutter/lib/src/widgets/_web_image_web.dart

```diff
@@ -0,0 +1,241 @@
+      'load',
+      (web.Event e) {
+        final int status = request.status;
+        final bool accepted = status >= 200 && status < 300;
+        final bool fileUri = status == 0; // file:// URIs have status of 0.
+        final bool notModified = status == 304;
+        final bool unknownRedirect = status > 307 && status < 400;
+        final bool success =
+            accepted || fileUri || notModified || unknownRedirect;
+
+        if (success) {
+          completer.complete(request);
+        } else {
+          completer.completeError(e);
+        }
+      }.toJS);
+
+  request.addEventListener(
+      'error', ((JSObject e) => completer.completeError(e)).toJS);
+
+  request.send();
+
+  return completer.future.then((_) {
+    return true;
+  }, onError: (_) {
+    return false;
+  });
+}
+
+/// Returns a widget which displays the [src] in an <img> tag.
+Widget createImgElementWidget(String src,
+        {Key? key,
+        ImageLoadingBuilder? loadingBuilder,
+        ImageFrameBuilder? frameBuilder,
+        ImageErrorWidgetBuilder? errorBuilder}) =>
+    _ImgElementImage(
+      src,
+      key: key,
+      loadingBuilder: loadingBuilder,
+      frameBuilder: frameBuilder,
+      errorBuilder: errorBuilder,
+    );
+
+class _ImgElementImage extends StatefulWidget {
+  const _ImgElementImage(
+    this.src, {
+    super.key,
+    this.loadingBuilder,
+    this.frameBuilder,
+    this.errorBuilder,
+  });
+
+  final String src;
+  final ImageLoadingBuilder? loadingBuilder;
+  final ImageFrameBuilder? frameBuilder;
+  final ImageErrorWidgetBuilder? errorBuilder;
+
+  @override
+  State<StatefulWidget> createState() => _ImgElementImageState();
+}
+
+class _ImgElementImageState extends State<_ImgElementImage> {
+  _ImgElementImageState() {
+    if (_registeredViewType == null) {
+      _register();
+    }
+  }
+
+  // Keeps track if this widget has already registered its view factories.
+  static String? _registeredViewType;
+
+  late _CachedImageInfo _cachedImage;
+
+  void _onCachedImageStateChange() {
+    // Rebuild when the image state changes.
+    setState(() {});
+  }
+
+  static const String _viewType = 'Browser__ImageElementType__';
+
+  static void _register() {
+    assert(_registeredViewType == null);
+    ui_web.platformViewRegistry.registerViewFactory(_viewType, (int viewId,
+        {Object? params}) {
+      final Map<Object?, Object?> paramsMap = params! as Map<Object?, Object?>;
+      final String src = paramsMap['src']! as String;
+      return _WebImageCache._getOrCreateCachedImage(src).image.cloneNode(true);
+    });
+  }
+
+  HtmlElementView? htmlElementView;
+
+  num? naturalWidth;
+  num? naturalHeight;
+
+  int? imgElementId;
+
+  @override
+  void initState() {
+    super.initState();
+    _cachedImage =
+        _WebImageCache._getOrCreateCachedImage(widget.src);
+    _cachedImage.state.addListener(_onCachedImageStateChange);
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
```

Comment:
Shouldn't we call `super.dispose()` last?

Per https://api.flutter.dev/flutter/widgets/State/dispose.html
`Implementations of this method should end with a call to the inherited method, as in super.dispose().`

Reply:
Done.